immix/llvm/plimmix_pass.cpp

```diff
@@ -49,31 +49,54 @@ namespace
   */
   void immixPassLogic(Module &M)
   {
+    auto isMain = false;
     for (auto FB = M.functions().begin(), FE = M.functions().end(); FB != FE; ++FB)
     {
       Function *FV = &*FB;
-      FV->setGC("plimmix");
+      if (FV->getName() == "main")
+      {
+        isMain = true;
+      }
+      if (!FV->getName().ends_with("visitorf@"))
+      {
+        FV->setGC("statepoint-example");
+      }
+      
     }
-    // auto gc_init_c = M.getOrInsertFunction("__gc_init_stackmap", Type::getVoidTy(M.getContext()));
-    auto immix_init_c = M.getOrInsertFunction("immix_gc_init", Type::getVoidTy(M.getContext()), PointerType::get(IntegerType::get(M.getContext(), 8), 0));
-    auto immix_init_f = cast<Function>(immix_init_c.getCallee());
-    immix_init_f->setLinkage(GlobalValue::LinkageTypes::ExternalLinkage);
-    SmallVector<Type *, 1> argTypes;
-    argTypes.push_back(PointerType::get(IntegerType::get(M.getContext(), 8), 0));
-    std::string symbol;
-    symbol += "_IMMIX_GC_MAP_";
-    symbol += M.getSourceFileName();
-    auto g = M.getOrInsertGlobal(symbol, Type::getInt8Ty(M.getContext()));
-    GlobalVariable *g_c = cast<GlobalVariable>(g);
-    g_c->setLinkage(GlobalValue::LinkageTypes::ExternalWeakLinkage);
-    // auto g = M.getNamedGlobal(symbol);
-    SmallVector<Value *, 1> assertArgs;
-    assertArgs.push_back(g);
-    Function *gc_init_f;
-    std::tie(gc_init_f, std::ignore) = createSanitizerCtorAndInitFunctions(M, "__gc_init_stackmap", "immix_gc_init", argTypes, assertArgs);
-    gc_init_f->setLinkage(GlobalValue::LinkageTypes::InternalLinkage);
-    // appendToCompilerUsed(M, gc_init_f);
-    appendToGlobalCtors(M, gc_init_f, 1000);
+    if (isMain)
+    {
+      // auto gc_init_c = M.getOrInsertFunction("__gc_init_stackmap", Type::getVoidTy(M.getContext()));
+      auto immix_init_c = M.getOrInsertFunction("immix_gc_init", Type::getVoidTy(M.getContext()), PointerType::get(IntegerType::get(M.getContext(), 8), 0));
```

Comment:
could you add some comments to explain the procedures here? As you changed almost the whole function implementation but haven't changed the function comment.